### ISSUE TITLE
Add ability to exclude ECMAScript features

### ISSUE THREAD

**Issue by migueldejuana** [eslint/eslint#12478]
**Please describe what the rule should do:**
The rule should allow us to exclude ECMAScript features
**What category of rule is this? (place an "X" next to just one item)**

[X ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
"rules": {
    "excludeFeature": ["SharedArrayBuffer", "Atomics"]
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**
ESLint allow us to config parseOptions with any ecmaVersion. There are some features that are not supported by browsers yet and would be nice to have a way to exclude those not supported features. 

**Are you willing to submit a pull request to implement this rule?**
No

**Comment by platinumazure** [comment]
Hi @migueldejuana, thanks for the issue.

You can already achieve this with a few rules:

* [`no-restricted-globals`](https://eslint.org/docs/rules/no-restricted-globals) lets you restrict the use of specific global variables, such as `SharedArrayBuffer` or `Atomics`.
* [`no-restricted-syntax`](https://eslint.org/docs/rules/no-restricted-syntax) lets you restrict the use of any particular ES syntax. You can use ESQuery to define arbitrarily complex syntax patterns you want to warn on.
* [`no-restricted-modules`](https://eslint.org/docs/rules/no-restricted-modules) and [`no-restricted-imports`](https://eslint.org/docs/rules/no-restricted-imports) let you restrict certain CommonJS modules or ES imports, respectively.

Please stop by our [Gitter chat](https://gitter.im/eslint/eslint) if you have questions on how to use each of those rules.

**Comment by mysticatea** [comment]
Or you can use [eslint-plugin-es](https://github.com/mysticatea/eslint-plugin-es#readme) that warns each ECMAScript feature.

**Comment by migueldejuana** [comment]
Thank you very much for your reply

**Comment by migueldejuana** [comment]
Enough with proposed solutions
